plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -129,66 +134,94 @@ public Network design(NetworkOffering offering, DeploymentPlan plan, Network use
 
     @Override
     public Network implement(Network network, NetworkOffering offering, DeployDestination dest, ReservationContext context) throws InsufficientVirtualNetworkCapacityException {
+        long networkId = network.getId();
+        network = _networkDao.acquireInLockTable(network.getId(), 1200);
+        if (network == null) {
+            throw new ConcurrentOperationException("Unable to acquire lock on network " + networkId);
+        }
 
-        assert (network.getState() == State.Implementing) : "Why are we implementing " + network;
+        NetworkVO implemented = null;
+        try {
+            assert (network.getState() == State.Implementing) : "Why are we implementing " + network;
+
+            long dcId = dest.getDataCenter().getId();
+            //Get physical network id
+            Long physicalNetworkId = network.getPhysicalNetworkId();
+            //Physical network id can be null in Guest Network in Basic zone, so locate the physical network
+            if (physicalNetworkId == null) {
+                physicalNetworkId = _networkModel.findPhysicalNetworkId(dcId, offering.getTags(), offering.getTrafficType());
+            }
+            implemented = new NetworkVO(network.getTrafficType(), network.getMode(), network.getBroadcastDomainType(), network.getNetworkOfferingId(), State.Allocated,
+                    network.getDataCenterId(), physicalNetworkId, offering.getRedundantRouter());
+            if (network.getGateway() != null) {
+                implemented.setGateway(network.getGateway());
+            }
+            if (network.getCidr() != null) {
+                implemented.setCidr(network.getCidr());
+            }
+            List<String[]> ipAddressRanges = new ArrayList<String[]>();
+            String virtualRouterIp = getVirtualRouterIP(network, ipAddressRanges);
+            String networkUuid = implemented.getUuid();
+            String tenantId = context.getDomain().getName() + "-" + context.getAccount().getAccountId();
+            String broadcastUriStr = networkUuid + "/" + virtualRouterIp;
+            implemented.setBroadcastUri(Networks.BroadcastDomainType.Vsp.toUri(broadcastUriStr));
+            implemented.setBroadcastDomainType(Networks.BroadcastDomainType.Vsp);
+            //Check if the network is associated to a VPC
+            Long vpcId = network.getVpcId();
+            boolean isVpc = (vpcId != null);
+            //Check owner of the Network
+            Domain networksDomain = _domainDao.findById(network.getDomainId());
+            //Get the Account details and find the type
+            AccountVO networksAccount = _accountDao.findById(network.getAccountId());
+            if (networksAccount.getType() == Account.ACCOUNT_TYPE_PROJECT) {
+                String errorMessage = "CS project support is not yet implemented in NuageVsp";
```

Comment:
It would be useful to add information about the account and device being configured in order to assist with operational debugging.